### PR TITLE
Keep Track of Build Statuses

### LINKED ISSUE #13: Notify When Builds Finish Processing
Use https://github.com/fastlane/watchbuild

### PR DESCRIPTION
**Pull request by johnf**
Keeps track of the last X builds and sends a message as they appear and with their status
Resolves #13

### CODE REVIEW

File: src/fetch_app_status.rb [hunk 1]
```diff
@@ -68,6 +72,18 @@ def get_version_info(app)
...
 end
 
+def get_build_info(app)
+  builds = app.get_builds.sort_by(&:uploaded_date).reverse[0, number_of_builds]
+
+  builds.map do |build|
+    {
+      "version" => build.version,
+      "uploaded_data" => build.uploaded_date,
+      "status" => build.processing_state,
```
Comment (rogerluan) [inline src/fetch_app_status.rb:82]: Can we switch to more modern syntax?

```suggestion
      version: build.version,
      uploaded_data: build.uploaded_date,
      status: build.processing_state,
```

File: src/poll-itc.js [hunk 2]
```diff
@@ -56,6 +59,29 @@ function _checkAppStatus(currentAppInfo) {
...
+                newBuildInfo[buildInfo.version] = buildInfo
+            } else {
+                console.log("No build change detected.")
+            }
+        })
+
+        db.set(buildInfoKey, newBuildInfo)
+    }
+
+    // Store latest build info in database
```
Comment (rogerluan) [inline src/poll-itc.js:84]: You probably meant to put this in the line above? haha
```suggestion
```

File: src/poll-itc.js [hunk 3]
```diff
@@ -56,6 +59,29 @@ function _checkAppStatus(currentAppInfo) {
...
+                newBuildInfo[buildInfo.version] = buildInfo
+            } else if (oldBuildInfo.status != buildInfo.status) {
+                poster.slackBuild(currentAppInfo, buildInfo)
+                newBuildInfo[buildInfo.version] = buildInfo
+            } else {
+                console.log("No build change detected.")
+            }
+        })
+
+        db.set(buildInfoKey, newBuildInfo)
```
Comment (rogerluan) [inline src/poll-itc.js:81]:
```suggestion
        // Store latest build info in database
        db.set(buildInfoKey, newBuildInfo)
```

**Review by rogerluan — CHANGES_REQUESTED** [review]

**Review by rogerluan — APPROVED** [review]
Thanks A LOT for this PR!! Your code LGTM. Looking forward to using this feature 🌟  Nicely done!

**Comment by johnf** [comment]
All good, thanks for merging!
Let me know when you do the next release if the docker workflow has any issues

**Comment by rogerluan** [comment]
I just published a release https://github.com/rogerluan/app-store-connect-notifier/releases/tag/v2.2.0 💪 I'll let you know if there're any issues 😁

**Comment by johnf** [comment]
You'll need to add two secrets , your docker hub username and password.
image

**Pull request merged**